Move top-level Shared WebExtension files to Shared/Extensions.

xeenon opened the pull request:
#### 890f5998df2e92c367379b909ef827c70c7000c2
<pre>
Move top-level Shared WebExtension files to Shared/Extensions.
<a href="https://bugs.webkit.org/show_bug.cgi?id=259839">https://bugs.webkit.org/show_bug.cgi?id=259839</a>
rdar://113413557

Reviewed by Brian Weinstein.

* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/Shared/Extensions/WebExtensionContextIdentifier.h: Renamed from Source/WebKit/Shared/WebExtensionContextIdentifier.h.
* Source/WebKit/Shared/Extensions/WebExtensionContextParameters.h: Renamed from Source/WebKit/Shared/WebExtensionContextParameters.h.
* Source/WebKit/Shared/Extensions/WebExtensionContextParameters.serialization.in: Renamed from Source/WebKit/Shared/WebExtensionContextParameters.serialization.in.
* Source/WebKit/Shared/Extensions/WebExtensionControllerIdentifier.h: Renamed from Source/WebKit/Shared/WebExtensionControllerIdentifier.h.
* Source/WebKit/Shared/Extensions/WebExtensionControllerParameters.h: Renamed from Source/WebKit/Shared/WebExtensionControllerParameters.h.
* Source/WebKit/Shared/Extensions/WebExtensionControllerParameters.serialization.in: Renamed from Source/WebKit/Shared/WebExtensionControllerParameters.serialization.in.
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: <a href="https://commits.webkit.org/266606@main">https://commits.webkit.org/266606@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/51a0ce3a45777323c1b2e4337fd83ef8e862de2d

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/14253 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/14567 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/14907 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/15989 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/13498 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/17080 "Built successfully") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/16/builds/14645 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/16170 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/14430 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/15/builds/14991 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/12091 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/16710 "Built successfully") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/39/builds/12275 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/19865 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/13353 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/13016 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/16220 "Passed tests") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/7/builds/13563 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/16/builds/14645 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/12846 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/17183 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/1696 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/13408 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

b-weinstein commented on Source/WebKit/DerivedSources-input.xcfilelist:
```diff
@@ -183,6 +183,8 @@ $(PROJECT_DIR)/Shared/Cocoa/WebCoreArgumentCodersCocoa.serialization.in
 $(PROJECT_DIR)/Shared/DisplayListArgumentCoders.serialization.in
 $(PROJECT_DIR)/Shared/EditorState.serialization.in
+$(PROJECT_DIR)/Shared/Extensions/WebExtensionContextParameters.serialization.in
+$(PROJECT_DIR)/Shared/Extensions/WebExtensionControllerParameters.serialization.in
```
Were these never here before? Was anything not working with these missing? Just out of curiousity.

xeenon replied:
Looks like not! This is auto updated by the build, so there must have been a bug when I added the files before.

xeenon committed:
Move top-level Shared WebExtension files to Shared/Extensions.
https://bugs.webkit.org/show_bug.cgi?id=259839
rdar://113413557

Reviewed by Brian Weinstein.

* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/Shared/Extensions/WebExtensionContextIdentifier.h: Renamed from Source/WebKit/Shared/WebExtensionContextIdentifier.h.
* Source/WebKit/Shared/Extensions/WebExtensionContextParameters.h: Renamed from Source/WebKit/Shared/WebExtensionContextParameters.h.
* Source/WebKit/Shared/Extensions/WebExtensionContextParameters.serialization.in: Renamed from Source/WebKit/Shared/WebExtensionContextParameters.serialization.in.
* Source/WebKit/Shared/Extensions/WebExtensionControllerIdentifier.h: Renamed from Source/WebKit/Shared/WebExtensionControllerIdentifier.h.
* Source/WebKit/Shared/Extensions/WebExtensionControllerParameters.h: Renamed from Source/WebKit/Shared/WebExtensionControllerParameters.h.
* Source/WebKit/Shared/Extensions/WebExtensionControllerParameters.serialization.in: Renamed from Source/WebKit/Shared/WebExtensionControllerParameters.serialization.in.
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: https://commits.webkit.org/266606@main
